homeassistant/components/homematicip_cloud/switch.py

```diff
@@ -69,16 +69,13 @@ async def async_setup_entry(
         elif isinstance(device, AsyncOpenCollector8Module):
             for channel in range(1, 9):
                 entities.append(HomematicipMultiSwitch(hap, device, channel=channel))
-        elif isinstance(device, AsyncHeatingSwitch2):
-            for channel in range(1, 3):
-                entities.append(HomematicipMultiSwitch(hap, device, channel=channel))
-        elif isinstance(device, AsyncMultiIOBox):
-            for channel in range(1, 3):
-                entities.append(HomematicipMultiSwitch(hap, device, channel=channel))
-        elif isinstance(device, AsyncPrintedCircuitBoardSwitch2):
-            for channel in range(1, 3):
-                entities.append(HomematicipMultiSwitch(hap, device, channel=channel))
-        elif isinstance(device, AsyncBrandSwitch2):
+        elif isinstance(
+            device,
+            AsyncBrandSwitch2
+            | AsyncPrintedCircuitBoardSwitch2
+            | AsyncHeatingSwitch2
+            | AsyncMultiIOBox,
```

Comment:
Not sure if there is _actually_ a performance difference. I slightly prefer using a tuple for `isinstance` instead. The `|` operator technically creates a `UnionType` first.

Reply:
Oh right, I read the ruff issue about this and I did execute the examples in that issue (since that was 3.10, maybe it improved)

![image](https://github.com/home-assistant/core/assets/7083755/6b913cc0-15eb-489b-85f9-f2f4dd7aba6d)

Reply:
Or make it a constant

Reply:
You mean make the tuple a constant?

Reply:
If you were going to keep the union a constant would be better.  Personally I prefer the tuple.